src/audio/eq_iir.c

```diff
@@ -594,16 +586,11 @@ static void eq_iir_cache(struct comp_dev *dev, int cmd)
 		trace_eq("inv");
 
 		dcache_invalidate_region(dev, sizeof(*dev));
-
-		cd = comp_get_drvdata(dev);
```

Comment:
comp_data should be retrieved after invalidation of dev, since we can potentially have the wrong pointer.

Reply:
Thanks, a new pull request with fix for this is on the way.